src/raftstore/store/fsm/peer.rs

```diff
@@ -120,136 +270,141 @@ impl<T, C> Store<T, C> {
             peer.raft_group.report_snapshot(to_peer_id, status)
         }
     }
+
+    pub fn collect_ready(&mut self, ctx: &mut PollContext<T, C>, proposals: &mut Vec<RegionProposal>) {
+        if self.has_ready && !self.stopped {
+            ctx.has_ready = true;
+            if let Some(p) = self.peer.take_apply_proposals() {
+                proposals.push(p);
+            }
+            self.peer.handle_raft_ready_append(ctx);
+        }
+    }
 }
 
-impl<T: Transport, C: PdClient> Store<T, C> {
-    pub fn register_raft_base_tick(&self, event_loop: &mut EventLoop<Self>) {
+impl<T: Transport, C: PdClient> PeerFsm<T, C> {
```

Comment:
why in different impl blocks? merge them together or split the functions into different blocks based on whether it is msg handle or tick handle.